references/cli/status.go

```diff
@@ -431,3 +453,89 @@ func printApplicationTree(c common.Args, cmd *cobra.Command, appName string, app
 	options.PrintResourceTree(cmd.OutOrStdout(), placements, currentRT, historyRTs)
 	return nil
 }
+
+// printRawApplication prints raw Application in yaml/json/jsonpath (without managedFields).
+func printRawApplication(ctx context.Context, c common.Args, format string, out io.Writer, ns, appName string) error {
+	var err error
+	app := &v1beta1.Application{}
+
+	k8sClient, err := c.GetClient()
+	if err != nil {
+		return fmt.Errorf("cannot get k8s client: %w", err)
+	}
+
+	err = k8sClient.Get(ctx, pkgtypes.NamespacedName{
+		Namespace: ns,
+		Name:      appName,
+	}, app)
+	if err != nil {
+		return fmt.Errorf("cannot get application %s in namespace %s: %w", appName, ns, err)
+	}
+
+	// Set GVK, we need it
+	// because the object returned from client.Get() has empty GVK
+	// (since the type info is inherent in the typed object, so GVK is empty)
+	app.SetGroupVersionKind(v1beta1.ApplicationKindVersionKind)
+	str, err := formatApplicationString(format, app)
+	if err != nil {
+		return err
+	}
+
+	_, err = out.Write([]byte(str))
+	return err
+}
+
+// formatApplicationString formats an Application to string in yaml/json/jsonpath for printing (without managedFields).
+//
+// format = "yaml" / "json" / "jsonpath={.field}"
+func formatApplicationString(format string, app *v1beta1.Application) (string, error) {
+	var ret string
+
+	// No, we don't want managedFields, get rid of it.
+	app.ManagedFields = nil
+
+	switch format {
+	case "yaml":
```

Comment:
extract it to be common util function. Maybe we need also format Component or Trait in the future.